tests/cnf/core/network/metallb/tests/metallb-nodeSelector.go

```diff
@@ -0,0 +1,274 @@
+
+				By("Validate Local Preference from Frr node0")
+				err = frr.ValidateLocalPref(frrPod0, 100, strings.ToLower(netparam.IPV4Family))
+				Expect(err).ToNot(HaveOccurred(), "Fail to validate local preference")
+
+				By("Validate Local Preference from Frr node1")
+				err = frr.ValidateLocalPref(frrPod1, 200, strings.ToLower(netparam.IPV4Family))
+				Expect(err).ToNot(HaveOccurred(), "Fail to validate local preference")
+
+				By(fmt.Sprintf("Validate BGP Community %s exists on received route prefix",
+					tsparams.NoAdvertiseCommunity))
+				bgpStatus, err := frr.GetBGPCommunityStatus(frrPod0, tsparams.NoAdvertiseCommunity,
+					strings.ToLower(netparam.IPV4Family))
+				Expect(err).ToNot(HaveOccurred(), "Failed to collect bgp community status")
+				Expect(len(bgpStatus.Routes)).To(Equal(1))
+
+				By(fmt.Sprintf("Validate BGP Community %s exists on received route prefix",
+					tsparams.CustomCommunity))
+				bgpStatus, err = frr.GetBGPCommunityStatus(frrPod1, tsparams.CustomCommunity,
+					strings.ToLower(netparam.IPV4Family))
+				Expect(err).ToNot(HaveOccurred(), "Failed to collect bgp community status")
+				Expect(len(bgpStatus.Routes)).To(Equal(1))
+			})
+	})
+})
+
+func createIPAddressPool(name string, ipPrefix []string) *metallb.IPAddressPoolBuilder {
+	ipAddressPool, err := metallb.NewIPAddressPoolBuilder(
+		APIClient,
+		name,
+		NetConfig.MlbOperatorNamespace,
+		[]string{fmt.Sprintf("%s-%s", ipPrefix[0], ipPrefix[1])}).Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create IPAddressPool")
+
+	return ipAddressPool
+}
+
+func setupTestCase(ipAddressPool1, ipAddressPool2 *metallb.IPAddressPoolBuilder,
+	frrk8sPods []*pod.Builder) (*pod.Builder, *pod.Builder) {
+	By("Creating two MetalLB service")
+
+	setupMetalLbService("service-1", netparam.IPV4Family, ipAddressPool1, "Cluster")
+	setupMetalLbService("service-2", netparam.IPV4Family, ipAddressPool2, "Cluster")
+
+	By("Creating nginx test pod on worker node 0")
+	setupNGNXPod(workerNodeList[0].Definition.Name)
+
+	By("Creating nginx test pod on worker node 1")
+	setupNGNXPod(workerNodeList[1].Definition.Name)
+
+	By("Creating External NAD for master FRR pods")
+	createExternalNad(tsparams.ExternalMacVlanNADName)
+
+	By("Creating static ip annotation for FRR master 0")
+
+	staticIPAnnotation0 := pod.StaticIPAnnotation(
+		tsparams.ExternalMacVlanNADName, []string{fmt.Sprintf("%s/%s", ipv4metalLbIPList[0], netparam.IPSubnet24)})
+
+	By("Creating static ip annotation for FRR master 1")
+
+	staticIPAnnotation1 := pod.StaticIPAnnotation(
+		tsparams.ExternalMacVlanNADName, []string{fmt.Sprintf("%s/%s", ipv4metalLbIPList[1], netparam.IPSubnet24)})
+
+	By("Creating MetalLb configMap for FRR master 0")
+
+	masterConfigMap0 := createConfigMap(tsparams.LocalBGPASN, ipv4NodeAddrList, false, false)
+
+	By("Creating MetalLb configMap for FRR master 1")
+
+	masterConfigMap1 := createConfigMap(tsparams.LocalBGPASN, ipv4NodeAddrList, false, false)
```

Comment:
Can we use the same configmap for the both masters?

Reply:
Yes removed the duplicate